lib/sshutils/scp/scp.go

```diff
@@ -517,9 +517,13 @@ func (cmd *command) receiveFile(st *state, fc newFileCmd, ch io.ReadWriter) erro
 		return trace.Errorf("unexpected file copy length: %v", n)
 	}
 
-	if err := cmd.FileSystem.Chmod(path, int(fc.Mode)); err != nil {
-		return trace.Wrap(err)
-	}
+  // Change the file permissions only when client requested it.
+  if cmd.Flags.PreserveAttrs {
+    if err := cmd.FileSystem.Chmod(path, int(fc.Mode)); err != nil {
+      return trace.Wrap(err)
+    }
+  }
```

Comment:
Indentation seems messed up in this block.

Reply:
Linter confirms. Fixing now.